src/rax.c

```diff
@@ -823,10 +823,10 @@ int raxGenericInsert(rax *rax, unsigned char *s, size_t len, void *data, void **
         trimmed->size = j;
         trimmed->iscompr = j > 1;
         trimmed->iskey = 0;
-        trimmed->isnull = 0;
+        trimmed->isnull = 1;
```

Comment:
If there is no key, the node also has no value, so the value of isnull is irrelevant. To initialize it to 1 is equally confusing so better leave it as 0, IMO.

```C
typedef struct raxNode {
    uint32_t iskey:1;     /* Does this node contain a key? */
    uint32_t isnull:1;    /* Associated value is NULL (don't store it). */
    ...
```

Reply:
I think you are correct. In fact, I was also very puzzled at the time. Why are iskey and isnull set to 0 in many initialization places. This is obviously a conflicting semantics. Of course, we can also think that iskey==true is not meaningless.

Reply:
Yes. I can guess that if isnull is not initialized, maybe there is a compiler warning.